### PR TITLE
feat(rxCharacterCount): Add a character counter directive and styles

### LINKED ISSUE #815: Character counter for text fields
We need to write styles in the framework for a character counter that counts down remaining characters.

Needed for:
- Billing https://jira.rax.io/browse/EOD-273
- Darrell's ACT Suspense Payment tool

Example: twitter
screen shot 2015-02-24 at 2 09 56 pm

### PR DESCRIPTION
**Pull request by parlarjb**
Closes #815

screen shot 2015-02-26 at 3 51 29 pm

screen shot 2015-02-26 at 3 51 39 pm

screen shot 2015-02-26 at 3 51 46 pm

### CODE REVIEW

**Comment by comeatmebro** [comment]
[Wraith UI diff](http://eb4d4842ea70055541c8-b6a3ba129b5eb76af13f4d17e4c1a53b.r51.cf5.rackcdn.com/821/gallery.html)

**Comment by jj4th** [comment]
Just popping in for a LGTM :)

**Comment by CITguy** [comment]
DEV - LGTM

**Comment by comeatmebro** [comment]
[Wraith UI diff](http://eb4d4842ea70055541c8-b6a3ba129b5eb76af13f4d17e4c1a53b.r51.cf5.rackcdn.com/821/gallery.html)

File: src/rxCharacterCount/templates/rxCommentBox.html [hunk 1]
```diff
@@ -0,0 +1 @@
+<div class="character-countdown" ng-class="{ 'near-limit': nearLimit, 'over-limit': overLimit }">{{ remaining }}</div>
```
Comment (megalord) [inline src/rxCharacterCount/templates/rxCommentBox.html]: This template is also defined inline in `src/rxCharacterCount/rxCharacterCount.js`.  Should this file be removed?

Comment (parlarjb) [inline src/rxCharacterCount/templates/rxCommentBox.html]: Gah, good catch

**Comment by comeatmebro** [comment]
[Wraith UI diff](http://eb4d4842ea70055541c8-b6a3ba129b5eb76af13f4d17e4c1a53b.r51.cf5.rackcdn.com/821/gallery.html)

**Pull request merged**